### PR TITLE
termite: add module

### PR DESCRIPTION
**Pull request by uvNikita**
Termite options were mostly autogenerated from this [config](https://github.com/thestinger/termite/blob/master/config). All options were made nullable to maintain the same default values as in Termite project.

### CODE REVIEW

File: modules/programs/termite.nix [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+        type = types.lines;
+        description = "Extra hints options that should be added to [hints] section.";
+      };
+    };
+  };
+
+  config = (
+    let
+      boolToString = v: if v then "true" else "false";
+      optionalBoolean = name: val: if val != null then "${name} = ${boolToString val}" else "";
```
Comment (rycee) [inline modules/programs/termite.nix]: This and the following three can use `optionalString`. For example

```
optionalBoolean = name: val: lib.optionalString (val != null) "${name} = ${boolToString val}";
```

File: modules/programs/termite.nix [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+        ${optionalString "cursor" cfg.cursorColor}
+        ${optionalString "cursor_foreground" cfg.cursorForegroundColor}
+        ${optionalString "foreground" cfg.foregroundColor}
+        ${optionalString "foregroundBold" cfg.foregroundBoldColor}
+        ${optionalString "highlight" cfg.highlightColor}
+
+        ${cfg.colorsExtra}
+
+        [hints]
+        ${optionalString "active_background " cfg.hintsActiveBackgroundColor}
```
Comment (rycee) [inline modules/programs/termite.nix]: All of these option strings have an extra space at the end. I guess that is unintentional?

**Comment by uvNikita** [comment]
Used `lib.optionalString` as you suggested and removed extra spaces.

**Comment by rycee** [comment]
Really cool new module, thanks! Rebased into master in c3be4c462907df02ff71a7f5d5cb1c6c4413cf8d.

**Pull request closed without merging**